test/microstrain_test.cpp

```diff
@@ -5,24 +5,43 @@
 
 namespace detail
 {
-    void warn_c_strings_equal(const char* string1, const char* string2)
+#ifdef MICROSTRAIN_TEST_USE_DOCTEST
+    void check_c_strings_equal(const char* actual, const char* expected)
     {
-        INFO(std::string(string1));
-        INFO(std::string(string2));
-        WARN_EQ(strcmp(string1, string2), 0);
+        const size_t safe_actual_size = strnlen(actual, MAX_STRING_LENGTH);
+        const size_t safe_expected_size = strnlen(expected, MAX_STRING_LENGTH);
+
+        INFO("Actual:   " << std::string(&actual[0], safe_actual_size));
+        INFO("Expected: " << std::string(&expected[0], safe_expected_size));
+        CHECK_EQ(strncmp(actual, expected, MAX_STRING_LENGTH), 0);
     }
 
-    void check_c_strings_equal(const char* string1, const char* string2)
+    void fail_if_position_out_of_bounds(const char *buffer, const size_t position)
     {
-        INFO(std::string(string1));
-        INFO(std::string(string2));
-        CHECK_EQ(strcmp(string1, string2), 0);
+        const size_t buffer_size = strnlen(buffer, MAX_STRING_LENGTH);
+
+        if (position > buffer_size)
+        {
+            //INFO("Position can't be greater than the buffer size");
+            INFO("Buffer size: " << buffer_size);
+            INFO("Position:    " << position);
+            FAIL("Position can't be greater than the buffer size");
+        }
     }
 
-    void require_c_strings_equal(const char* string1, const char* string2)
+	void check_buffer_terminated(const char *buffer, const size_t position)
     {
-        INFO(std::string(string1));
-        INFO(std::string(string2));
-        REQUIRE_EQ(strcmp(string1, string2), 0);
+        // TODO: Maybe we could add the functionality to pass -1 for the position to automatically
+        //       check the back of the buffer?
+        fail_if_position_out_of_bounds(buffer, position);
```

Comment:
I'd be ok with allowing negative indices to count back from the end. But I'm not sure how useful that is, given none of the tests write anywhere near 1024 characters.